### PR TITLE
Oxidise

### PR DESCRIPTION
**Pull request by fables-tales**

### CODE REVIEW

**Comment by lopopolo** [comment]
Hi @penelopezone 💎 

I came across this branch on your Twitter and I am _very interested_ in seeing how I can help so I can use the parser in [Artichoke](https://github.com/artichoke/artichoke), which aspires to be a Ruby implemented in Rust.

I'm hoping that this branch might result in a Rust crate that Artichoke can depend on.

See: https://github.com/artichoke/artichoke/issues/139

Cheers,
Ryan

File: native/src/format.rs [hunk 1]
```diff
@@ -0,0 +1,1885 @@
...
+    });
+
+    ps.emit_end();
+    ps.emit_newline();
+}
+
+pub fn inner_format_params(ps: &mut ParserState, params: Params) {
+    let non_null_positions = params.non_null_positions();
+    //def foo(a, b=nil, *args, d, e:, **kwargs, &blk)
+    //        ^  ^___^  ^___^  ^  ^    ^_____^   ^
```
Comment (mikker) [inline native/src/format.rs]: `^___^` indeed

**Comment by penelopezone** [comment]
This pull request was opened(!!) on the 14th of July, 2019. It's now in a state where the rust version of this is able to run on all of the code in rspec/rspec-core, and not break any test, so we know it's a reasonable platform on top of which to build.

Going to resume working on master, and replace the ruby version with the rust version entirely. Merging and closing for now.

**Comment by mcmire** [comment]
This is cool! It's been fun to watch you work on this. I'm happy you're continuing to make progress on it!

**Comment by twe4ked** [comment]
Awesome to see this merged! 🎉

**Pull request merged**